resim/msg/fuzz_helpers.hh

```diff
@@ -147,6 +149,40 @@ Detection3D random_element(TypeTag<Detection3D> /*unused*/, InOut<Rng> rng) {
   return result;
 }
 
+template <typename Rng>
+NavSatFix random_element(TypeTag<NavSatFix> /*unused*/, InOut<Rng> rng) {
+  constexpr std::array STATUSES = {
+      NavSatFix::STATUS_NO_FIX,
+      NavSatFix::STATUS_FIX,
+      NavSatFix::STATUS_SBAS_FIX,
+      NavSatFix::STATUS_GBAS_FIX,
+  };
+  constexpr std::array COVARIANCES = {
+      NavSatFix::COVARIANCE_TYPE_UNKNOWN,
+      NavSatFix::COVARIANCE_TYPE_APPROXIMATED,
+      NavSatFix::COVARIANCE_TYPE_DIAGONAL_KNOWN,
+      NavSatFix::COVARIANCE_TYPE_KNOWN,
+  };
+
+  NavSatFix result;
+  result.mutable_header()->CopyFrom(random_element<Header>(rng));
+  result.set_status(
+      STATUSES.at(random_element<std::size_t>(rng) % STATUSES.size()));
+  result.set_latitude(random_element<double>(rng));
+  result.set_longitude(random_element<double>(rng));
+  result.set_altitude_m(random_element<double>(rng));
+
+  constexpr int COV_DIM = 9;
+  for (int ii = 0; ii < COV_DIM; ++ii) {
+    // Absolutely not a valid covariance, but we only care about data
+    // preservation.
```

Comment:
You could do the old $\Sigma = X^TX$ if you want to be all fancy here... (But as a side note, I actually think the root matrix of a positive semi-definite matrix is often more natural as a "type". Wish we could change conventions on these things.)